src/components/Executions/ExecutionDetails/ExecutionDetailsAppBarContent.tsx

```diff
@@ -127,20 +128,26 @@ export const ExecutionDetailsAppBarContent: React.FC<{
         <TerminateExecutionButton className={styles.actionButton} />
     ) : isTerminal ? (
         <>
-            <Button
-                variant="outlined"
-                color="primary"
-                disabled={recovering}
-                className={classnames(
-                    styles.actionButton,
-                    commonStyles.buttonWhiteOutlined
-                )}
-                onClick={onClickRecover}
-                size="small"
-            >
-                Recover
-                {recovering && <ButtonCircularProgress />}
-            </Button>
+            {[
+                WorkflowExecutionPhase.FAILED,
+                WorkflowExecutionPhase.ABORTED,
+                WorkflowExecutionPhase.TIMED_OUT
+            ].includes(phase) && (
```

Comment:
Could we change this to `const isRecoverVisible = React.useMemo(() => [WorkflowExecutionPhase.FAILED,WorkflowExecutionPhase.ABORTED,WorkflowExecutionPhase.TIMED_OUT].includes(phase), [phase]);` and use that variable in dom?
Just my suggestion, ;)

Reply:
Nice catch, just pushed the fix.